asserts/snap_asserts.go

```diff
@@ -316,3 +316,134 @@ func assembleSnapRevision(assert assertionBase) (Assertion, error) {
 		timestamp:     timestamp,
 	}, nil
 }
+
+// SnapValidation holds a validation assertion, describing that a combination of
+// (snap-id, snap-revision, approved-snap-id, approved-revision, series)
+// has been validated by testing.
+
+type Validation struct {
+	assertionBase
+	valid     bool
+	timestamp time.Time
+}
+
+// Series returns the series for which the validation holds.
+func (validation *Validation) Series() string {
+	return validation.HeaderString("series")
+}
+
+// SnapId returns the ID of the gating snap.
+func (validation *Validation) SnapID() string {
+	return validation.HeaderString("snap-id")
+}
+
+// ApprovedSnapId returns the ID of the gated snap.
+func (validation *Validation) ApprovedSnapID() string {
+	return validation.HeaderString("approved-snap-id")
+}
+
+// ApprovedSnapRevision returns the revision of the gated snap.
+func (validation *Validation) ApprovedSnapRevision() string {
+	return validation.HeaderString("approved-snap-revision")
+}
+
+// IsValid returns true if the validation is marked as valid
+func (validation *Validation) IsValid() bool {
+	return validation.valid
+}
+
+// Timestamp returns the timestamp of assertion creation
```

Comment:
"Timestamp returns the time when the validation was issued." ... just for consistency with other assertion types.